chain/xrp/client/client_test.go

```diff
@@ -0,0 +1,698 @@
+						},
+						"PreviousTxnID": "FACEF612B2D8EA190AED5A576E9236C76FC19BC791139FC4CD99C1D7246167BF",
+						"PreviousTxnLgrSeq": 90659175
+					  }
+					},
+					{
+					  "ModifiedNode": {
+						"LedgerEntryType": "AccountRoot",
+						"LedgerIndex": "BF1F2A23D614916E3C6ED2DCC389468CFA09045BEDB54C71A05C5E94EA6C6CFE",
+						"PreviousTxnID": "23132EB4C93F01A84AD3DD5132FFA0EB1BBD4F7F18C4651A38E1F75998B39D90",
+						"PreviousTxnLgrSeq": 90658052
+					  }
+					},
+					{
+					  "ModifiedNode": {
+						"FinalFields": {
+						  "Flags": 0,
+						  "IndexNext": "1",
+						  "IndexPrevious": "2391",
+						  "Owner": "rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq",
+						  "RootIndex": "D7AC7D74720E29A563100F2B494BADB198F8A9E9FA46F57AE07123151E0DFA7A"
+						},
+						"LedgerEntryType": "DirectoryNode",
+						"LedgerIndex": "D7AC7D74720E29A563100F2B494BADB198F8A9E9FA46F57AE07123151E0DFA7A",
+						"PreviousFields": {
+						  "IndexPrevious": "2390"
+						}
+					  }
+					}
+				  ],
+				  "TransactionIndex": 1,
+				  "TransactionResult": "tesSUCCESS"
+				},
+				"validated": true,
+				"date": 779303540,
+				"ledger_index": 90659219,
+				"inLedger": 90659219,
+				"status": "success"
+			  },
+			  "warnings": [
+				{
+				  "id": 2001,
+				  "message": "This is a clio server. clio only serves validated data. If you want to talk to rippled, include 'ledger_index':'current' in your request"
+				}
+			  ]
+			}`,
+			ledgerResp: xrpClient.LedgerResponse{
+				Result: xrpClient.LedgerResult{
+					Ledger: xrpClient.LedgerInfo{
+						Closed:      false,
+						LedgerIndex: "91225188",
+						ParentHash:  "3E54FF795235548F8B62078F9CE5B5427D7B86BB73571C5CBD9044E171842218",
+					},
+					LedgerCurrentIndex: 91225188,
+					Validated:          false,
+					Status:             "success",
+				},
+			},
+			err: "",
+		},
+	}
+
+	for testNo, vector := range vectors {
+
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			var reqBody map[string]interface{}
+			json.NewDecoder(r.Body).Decode(&reqBody)
+
+			method := reqBody["method"].(string)
+			if method == "tx" {
+				// Respond with AccountInfoResponse
+				w.Write([]byte(vector.txResp))
+			} else if method == "ledger" {
+				// Respond with LedgerResponse
+				json.NewEncoder(w).Encode(vector.ledgerResp)
+			} else {
+				t.Errorf("unexpected method: %s", method)
+			}
+		}))
+		defer server.Close()
+
+		if token, ok := vector.asset.(*xc.TokenAssetConfig); ok {
+			token.ChainConfig = &xc.ChainConfig{
+				URL:   server.URL,
+				Chain: "XRP",
+			}
+		} else {
+			vector.asset.(*xc.ChainConfig).URL = server.URL
+		}
+
+		client, _ := xrpClient.NewClient(&xc.ChainConfig{URL: server.URL})
+		txInfo, err := client.FetchTxInfo(context.Background(), xc.TxHash(vector.txHash))
+
+		if vector.err != "" {
+			require.Equal(t, xc.LegacyTxInfo{}, txInfo)
+			require.ErrorContains(t, err, vector.err)
+		} else {
+			if testNo == 0 {
```

Comment:
don't do this, should be able to just have the expected result in the test case.

Reply:
It won't work like that.
As I am passing txResponse as JSON string.
.
If you don't like testCase == 0 or 1. I can do two tests.

Reply:
Or to create another JSON string object with expected results but imo it won't look better.

Reply:
I don't understand.  In both cases "0" and "1", you're comparing only the resulting `client.TxInfo` type.  So in your test case you should have something like:

```go
{
 // ...
  expectedTxInfo: client.TxInfo{
   // expected fields
  }
}
```

Doesn't really matter how the input json is setup.